### PR TITLE
dev

### PR DESCRIPTION
**Pull request by lutianzhou001**

### CODE REVIEW

File: api/GetBlockByHeightInHex.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    "required": [
+        "Height"
+    ],
+    "additionalProperties": false
+}
+```
+
+## Output
+
+``` json
```
Comment (vang1ong7ang) [inline api/GetBlockByHeightInHex.md]: remove space

File: api/GetContractByHashHeightInHex.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            "type": "number",
+            "minumum": 0,
+            "description": "height"
+        },
+        "Hash": {
+            "type": "string",
+            "minLength": 64,
+            "maxLength": 64,
+            "pattern": "[0-9a-f]{64}",
+            "description": "sha256^2 hash in big endian"
```
Comment (vang1ong7ang) [inline api/GetContractByHashHeightInHex.md]: use script hash instead of sha256^2 hash

File: api/GetContractByHashLEHeightInHex.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+            "type": "number",
+            "minumum": 0,
+            "description": "height"
+        },
+        "Hash": {
+            "type": "string",
+            "minLength": 64,
+            "maxLength": 64,
+            "pattern": "[0-9a-f]{64}",
+            "description": "sha256^2 hash in little endian"
```
Comment (vang1ong7ang) [inline api/GetContractByHashLEHeightInHex.md]: Use script hash

File: api/GetHashByHeightInHex.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+}
+```
+
+## Output
+
+``` json
+{
+    "type": "string",
+    "pattern": "[0-9a-f]+",
+    "description": "binary block data encoded in hex format"
```
Comment (vang1ong7ang) [inline api/GetHashByHeightInHex.md]: Fixit

File: api/GetHashByHeightInHex.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    "additionalProperties": false
+}
+```
+
+## Output
+
+``` json
+{
+    "type": "string",
+    "pattern": "[0-9a-f]+",
```
Comment (vang1ong7ang) [inline api/GetHashByHeightInHex.md]: ［0-9a-f]{64}

File: api/GetRawTransactionByHashInJSON.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
+# GetRawTransactionByHashInJSON
```
Comment (vang1ong7ang) [inline api/GetRawTransactionByHashInJSON.md]: Change the name to 

GetTransactionByHashInJSON

File: api/GetRawTransactionByHashLEhInJSON.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
+# GetRawTransactionByHashLEInJSON
+
```
Comment (vang1ong7ang) [inline api/GetRawTransactionByHashLEhInJSON.md]: Same as before

File: api/GetStorageByHashHeightInHex.md [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+            "type": "number",
+            "minumum": 0,
+            "description": "height"
+        },
+        "Hash": {
+            "type": "string",
+            "minLength": 64,
+            "maxLength": 64,
+            "pattern": "[0-9a-f]{64}",
+            "description": "sha256^2 hash in big endian"
```
Comment (vang1ong7ang) [inline api/GetStorageByHashHeightInHex.md]: sha256^2 hash of `script hash of contract + key`

File: api/GetStorageByHashLEHeightInHex.md [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+            "type": "number",
+            "minumum": 0,
+            "description": "height"
+        },
+        "Hash": {
+            "type": "string",
+            "minLength": 64,
+            "maxLength": 64,
+            "pattern": "[0-9a-f]{64}",
+            "description": "sha256^2 hash in little endian"
```
Comment (vang1ong7ang) [inline api/GetStorageByHashLEHeightInHex.md]: Same as before

Comment (vang1ong7ang) [reply api/GetStorageByHashLEHeightInHex.md]: sha256^2 hash of `contract script hash (little endian) + storage key`

File: api/GetBlockByHashInHex.md [hunk 10]
```diff
@@ -35,4 +35,7 @@ get block by hash in hex
 
 ## Examples
 
-TODO
\ No newline at end of file
+```sh
+curl https://mainnet.neophora.io -d '{"id":1,"jsonrpc":"2.0","method":"GetBlockByHashInHex","params":{"Hash":"e66c8bf88a93abf37c8b840860dbed67daf70a9a72ae31f977fc0c68e4be8aed"}}'
```
Comment (vang1ong7ang) [inline api/GetBlockByHashInHex.md]: https://example.neophora.io

File: api/GetBlockByHeightInHex.md (hunk 1 above)
Comment (lutianzhou001) [reply api/GetBlockByHeightInHex.md]: done

File: api/GetContractByHashHeightInHex.md (hunk 2 above)
Comment (lutianzhou001) [reply api/GetContractByHashHeightInHex.md]: done

File: api/GetContractByHashLEHeightInHex.md (hunk 3 above)
Comment (lutianzhou001) [reply api/GetContractByHashLEHeightInHex.md]: done

File: api/GetHashByHeightInHex.md (hunk 4 above)
Comment (lutianzhou001) [reply api/GetHashByHeightInHex.md]: done

File: api/GetHashByHeightInHex.md (hunk 5 above)
Comment (lutianzhou001) [reply api/GetHashByHeightInHex.md]: done

File: api/GetRawTransactionByHashInJSON.md (hunk 6 above)
Comment (lutianzhou001) [reply api/GetRawTransactionByHashInJSON.md]: done

File: api/GetRawTransactionByHashLEhInJSON.md (hunk 7 above)
Comment (lutianzhou001) [reply api/GetRawTransactionByHashLEhInJSON.md]: done

File: api/GetStorageByHashHeightInHex.md (hunk 8 above)
Comment (lutianzhou001) [reply api/GetStorageByHashHeightInHex.md]: done

File: api/GetStorageByHashLEHeightInHex.md (hunk 9 above)
Comment (lutianzhou001) [reply api/GetStorageByHashLEHeightInHex.md]: done

**Pull request merged**